XUtils.h

```diff
@@ -76,4 +77,13 @@ char* xStrndup(const char* str, size_t len) ATTR_NONNULL ATTR_MALLOC;
 ssize_t xReadfile(const char* pathname, void* buffer, size_t count);
 ssize_t xReadfileat(openat_arg_t dirfd, const char* pathname, void* buffer, size_t count);
 
+#ifdef HAVE_BUILTIN_CLZ
+static inline unsigned int prevPowerOf2(unsigned int x) {
+   return ((x <= 0) ? 0 :
+           (1U << (sizeof(x) * CHAR_BIT - 1 - __builtin_clz(x))));
```

Comment:
No need to press everything in one line.

Also check for `x<2` to directly return `0` on shortcut route.

Reply:
Added a comment to clarify the behavior of prevPowerOf2().
There's no problem with `(x<=0)` check. It's there to avoid undefined behavior of `__builtin_clz()`. Other things work as expected.